### PR TITLE
Return blob from image-only type

### PR DESCRIPTION
**Pull request by vendramini**
I have to create this function:
```javascript
/**
     * Display snapshot image.
     * @private
     */
    displaySnapshot() {
        // hide preview video
        this.mediaElement.style.display = 'none';

        // show the snapshot
        this.player.recordCanvas.show();

        // stop recording
        this.stop();
    }
```
Because the process to transform into blob is async, and it was not firing correctly the data.

Usage:
```javascript
plugins: {
        record: {
            image: true, // return base64 by default from canvas
            // image: {
            //   returnType: 'blob' // return blob from canvas
            // },
            debug: true
        }
    },
```

### CODE REVIEW

**Comment by vendramini** [comment]
omg my previous PR comes together.

Sorry, I'm not a git expert. Do I did something wrong? I'd created a new branch then a new PR with it.

File: CHANGES.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 videojs-record changelog
 ========================
+3.8.1 - 2019/09/10
+------------------
+- Added `Event.RETRY` only in picture case.
+
```
Comment (thijstriemstra) [inline CHANGES.md]: there are now 2 changelog entries for 3.8.1, please reduce it to 1.

File: src/js/videojs.record.js [hunk 2]
```diff
@@ -1436,23 +1436,48 @@ class Record extends Plugin {
...
     /**
-     * Create and display snapshot image.
+     * Create snapshot image.
      * @private
      */
     createSnapshot() {
         this.captureFrame().then((result) => {
-            // turn the canvas data into base64 data with a PNG header
-            this.player.recordedData = result.toDataURL('image/png');
+            if (this.player.options_.plugins.record.image.returnType === 'blob')
```
Comment (vendramini) [inline src/js/videojs.record.js]: @thijstriemstra should I do a check if `.image` is an object? I though know, maybe it could be a boolean?

**Comment by vendramini** [comment]
Done. Please check.

**Comment by thijstriemstra** [comment]
@vendramini I had some spare time and added some options to support this feature, see attached patch: 
[image-blob.patch.txt](https://github.com/collab-project/videojs-record/files/3622855/image-blob.patch.txt)

Apply this to the master branch using `git patch -p0 < blob.patch.txt`. Can you update your PR with this and implement the rest (e.g. documentation, defining actual vars in `loadOptions` etc).

**Comment by vendramini** [comment]
@thijstriemstra just to you know, I'm very busy now. I'll back here soon, probably next week. Sorry the delay.

**Comment by thijstriemstra** [comment]
see #403

**Pull request closed without merging**